src/main/cluster-detectors/distribution-detector.ts

```diff
@@ -56,12 +56,12 @@ export class DistributionDetector extends BaseClusterDetector {
       return { value: "docker-desktop", accuracy: 80};
     }
 
-    if (this.isCustom()) {
-      return { value: "custom", accuracy: 10};
+    if (this.isCustom() && await this.isOpenshift()) {
```

Comment:
Why checking both here?

Reply:
It's a performance optimisation, we don't want to check Openshift (which is an API call) unless we know it's worth it.

Reply:
Ah fair, would it then be better to do the following?

```typescript
if (this.isCustom()) {
  if (await this.isOpenshift()) {
    return { value: "openshift", accuracy: 90 };
  }

  return { value: "custom", accuracy: 10 };
}
```

Reply:
I'm not sure if that's more readable.. involves more nesting.

Reply:
okay, and the custom check is pretty easy anyway.